ports/samd/machine_uart.c

```diff
@@ -479,12 +486,18 @@ static void mp_machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args,
 }
 
 static mp_obj_t mp_machine_uart_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
+    mp_arg_check_num(n_args, n_kw, 0, MP_OBJ_FUN_ARGS_MAX, true);
```

Comment:
Maybe worth doing `MICROPY_HW_DEFAULT_UART_ID < 0 ? 1 : 0` for the third argument, so if there's no default UART ID for the given board, the error message is slightly better (it'll say that the constructor is missing a required positional argument).

Reply:
Good suggestion. Done.

Reply:
To have the same behavior for I2C a mp_arg_check_num() has to be added. That increases the code size by 20 bytes. Even without that, the missing argument is flagged, only with a different error message of `ValueError: I2C(-1) doesn't exist` instead of `TypeError: argument num/types mismatch`.

Reply:
For I2C you can do the following at zero code size cost:
```c
#if MICROPY_HW_DEFAULT_I2C_ID < 0            
{ MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_INT, {.u_int = -1} },             
#else
{ MP_QSTR_id, MP_ARG_INT, {.u_int = MICROPY_HW_DEFAULT_I2C_ID} },
#endif
```

Reply:
That's a good hint. Done.